### ISSUE TITLE
Issue writing Aimbot

### ISSUE THREAD

**Issue by ZorCoder** [MarkHC/CSGOSimple#85]
Hi, im trying to make an aimbot with your base, but I can't seem to find to transfer my old code to the new one, in this base viewangle is a QAngle, but for example im my old cheat it was a Vector. I can seem to be able to transfer everything. Please help.

(Maybe give us some math's related to this)

**Comment by M-T3K** [comment]
Hey ZorCoder, 

I suggest you take a look at the QAngle.hpp file (Located at \valve_sdk\math). In it, you will realize that QAngle is essentially a Vector, and it includes a function / method called  QAngle(float X, float Y, float Z). Therefore, porting your previous Vector over to a QAngle should not be too much of an issue. QAngle also comes packed with several other functions that you should check as well.

**Comment by ZorCoder** [comment]
Thanks. It took some time but I fixed it. but now Im having trouble registering if the mouse key is held, which sadly doesn't work. Do you maybe know something about that?

**Comment by M-T3K** [comment]
Well, there are several ways to do this. Since you haven't included any source on how you are approaching it, I'll let you know the way I have done it:

I used GetAsyncKeyState(i & 0x8000), where i represents the Integer of the key pressed, like VK_LBUTTON.  Make sure you are using GetAsyncKeyState instead of GetKeyState, I believe that could be your issue.

**Comment by M-T3K** [comment]
@ZorCoder  I am not exactly sure aout what the problem can be, because Access Violation Exceptions are usually system specific.  Maybe try to use a try / catch and compiling with /EHa. Otherwise, take a look at this from the Microsoft Documentation:
https://msdn.microsoft.com/en-us/library/6decc55h.aspx

Otherwise, with just that one line I really don't know what is going on. If none of these help you, maybe try an alternative way of doing it.

**Comment by iWonAMathsDebate** [comment]
Why the hell bother using GetAsyncKeyState? There is already an implemented GetKeyState function in the base. Take a look at main.cpp and see for yourself where the keybind to menu is created.

**Comment by M-T3K** [comment]
@iWonAMathsDebate Yeah, I know that. My comment was like almost 4 months ago now. Even though I don't really use this base anymore ( I was just experimenting with it) I did find out about the InputSystem, which I really like, actually.
